game-core/src/test/java/games/strategy/triplea/delegate/battle/steps/BattleStepsTest.java

```diff
@@ -0,0 +1,2199 @@
+import static games.strategy.triplea.delegate.battle.steps.BattleSteps.FIRE;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.GamePlayer;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.engine.data.properties.GameProperties;
+import games.strategy.triplea.Constants;
+import games.strategy.triplea.attachments.TechAttachment;
+import games.strategy.triplea.attachments.UnitAttachment;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class BattleStepsTest {
+
+  @Mock GameData mockGameData;
+  @Mock GameProperties mockGameProperties;
+  @Mock Function<Collection<Unit>, Collection<Unit>> mockGetDependentUnits;
+  @Mock Supplier<Collection<Territory>> getAttackerRetreatTerritories;
+
+  @Mock
+  BiFunction<GamePlayer, Collection<Unit>, Collection<Territory>> getEmptyOrFriendlySeaNeighbors;
+
+  @Mock Territory mockBattleSite;
+  @Mock GamePlayer attacker;
+  @Mock GamePlayer defender;
+  @Mock TechAttachment techAttachment;
+  @Mock Unit unit1;
+  @Mock UnitType mockUnit1Type;
+  @Mock UnitAttachment mockUnit1Attachment;
+  @Mock Unit unit2;
+  @Mock UnitType mockUnit2Type;
+  @Mock UnitAttachment mockUnit2Attachment;
+  @Mock Unit unit3;
+  @Mock UnitType mockUnit3Type;
+  @Mock UnitAttachment mockUnit3Attachment;
+  @Mock Unit unit4;
+  @Mock UnitType mockUnit4Type;
+  @Mock UnitAttachment mockUnit4Attachment;
+
+  @BeforeEach
+  void setupMocks() {
+    when(mockGameData.getProperties()).thenReturn(mockGameProperties);
+  }
+
+  private void players() {
+    when(attacker.getName()).thenReturn("mockAttacker");
+    when(defender.getName()).thenReturn("mockDefender");
+  }
+
+  private void unit1() {
+    when(unit1.getType()).thenReturn(mockUnit1Type);
+    when(mockUnit1Type.getAttachment(anyString())).thenReturn(mockUnit1Attachment);
+  }
+
+  private void unit2() {
+    when(unit2.getType()).thenReturn(mockUnit2Type);
+    when(mockUnit2Type.getAttachment(anyString())).thenReturn(mockUnit2Attachment);
+  }
+
+  private void unit3() {
```

Comment:
The helper setup methods are good things :+1: 
Explicit input/output for them would not be bad though, as mentioned in another comment.
Another way to do the explicit input/output, for this method, as an example, woudl be:

```
private static void givneUnitHasAttachment(Unit unit, UnitAttachment unitAttachment) {
  UnitAttachment unitAttachment = mock(UnitAttachment.class);
   when(unit.getType).thenReturn(unitAttachment);
   when(unitAttachment.getAttachment(anyString()).thenReturn(unitAttachment);
}
```

Then as usage:
```
givenUnitHasAttachment(unit3, mockUnit3Attachment);
```
In the above you can see explicitly which variables are being configured.


These should complement well, how tests are done, for example:
```
Unit aaUnit = givenUnitWithAttachment(unitAttachment1);
when(unitAttachment.getTypeAa()).thenReturn("AntiAirGun");

Unit notAaUnit = givenUnitWithAttachment(unitAttachment2);
when(unitAttachment.getTypeAa()).thenReturn("");
```
In the above, later when you use `aaUnit` and `notAaUnit` it'll be a bit more explicit how they were configured and why we have expectations for one unit or another to be in an output list.


If you go the route of breaking up BattleStep, then it'll potentially start to make sense to have even more involved helper methods for the test, ie:
```
Unit aaUnit = givenUnitWithAaType("AntiAirGun");
Unit notAaUnit = givenUnitWithAaType("");
:
:
```